client/src/components/Badge/tests/Badge-test.js

```diff
@@ -1,32 +1,39 @@
-/* global jest, describe, beforeEach, it, expect */
+/* global jest, describe, it, expect */
 
-jest.unmock('../Badge');
-
-// FYI: Changing these to import statements broke jest's automocking
 import React from 'react';
 import ReactTestUtils from 'react-addons-test-utils';
 import Badge from '../Badge';
 
 describe('Badge', () => {
-  let props = null;
-
-  beforeEach(() => {
-    props = {
-      status: null,
-      message: '',
-      className: '',
-    };
-  });
-
   describe('render()', () => {
     let badge = null;
 
-    it('shoudl return null if status is empty', () => {
+    it('should return null if status is empty', () => {
+      badge = ReactTestUtils.renderIntoDocument(
+        <Badge
+          status={null}
+          message=""
+          className=""
+        />
+      );
+
+      expect(badge.render()).toBeNull();
+    });
+
+    it('should return a Bootstrap style badge when valid', () => {
       badge = ReactTestUtils.renderIntoDocument(
-        <Badge {...props} />
+        <Badge
+          status="success"
+          message="Hello world"
+          className="customclass"
+        />
       );
 
-      expect(badge).toBe(null);
+      expect(badge.render()).toEqual(
+        <span className="customclass badge badge-success badge-pill">
+          Hello world
+        </span>
+      );
```

Comment:
I prefer cherry picking parts that you truly expect, than a whole dump of a big object...
so test that `badge-success` class exists
test that `Hello world` is in the `textContent`

That way unit tests don't ALWAYS break when we tweak the component

Reply:
Sure. I’d argue the opposite in that changes to a component should be reflected in the tests too, but I agree that partial assertions would be totally fine

Reply:
Updated =)

Reply:
Yeah, I'd argue with breaking that test down even more `it()` statements :D
That way you test each feature atomically, this refactor is good to go though.

Reply:
Good counter! Thanks :-)